core/feed_protocol/acquisition.py

```diff
@@ -590,11 +589,12 @@ def single_entry_loans_feed(
 
         if isinstance(entry, WorkEntry) and entry.computed:
             return cls.entry_as_response(entry, **response_kwargs)
-        elif entry is None:
-            return None
+        elif isinstance(entry, OPDSMessage):
+            return ProblemDetail(
```

Comment:
It looks like this new case isn't covered by a test, and it seems like the original code doesn't return a problem detail in this case, but converts the `OPDSMessage` into an error response?

Can you write up a test for this and verify that we aren't changing the behavior of the single item loans feed here?

Reply:
I think either: 
- we need to handle this at the serialization step, so that we get the same XML response we used to get in the case of an OPDS 1 feed
- Or check with the mobile apps and see how returning a problem detail here instead of a feed will be handled

Since our mobile apps are the primary consumer of this feed still, we need to make sure they are in sync with any changes that get made to the responses.

Reply:
We can return to the XML format easily, will that be an acceptable behaviour when the client is expecting JSON though?
Re: ProblemDetail: The same method also returns problem details [in different circumstances](https://github.com/ThePalaceProject/circulation/pull/1340/files/8c4941b94515b085a85b482a88d366642315b588..508b81a66336ff8bd846c90401df0ce490ba6391#diff-915f371a570738a5b9970eca750c31619010b101b4ca87634fcbcaf01649c078R561) so a problem detail should be already handled by the client side.

Reply:
Ok, I reverted to protocol specific data, OPDS1 will be XML and OPDS2 will be JSON.